packages/aws-cdk-lib/aws-cloudfront-origins/lib/s3-bucket-origin.ts

```diff
@@ -0,0 +1,248 @@
+          }
+        }
+
+        const originBindConfig = this._bind(scope, options);
+
+        // Update configuration to set OriginControlAccessId property
+        return {
+          ...originBindConfig,
+          originProperty: {
+            ...originBindConfig.originProperty!,
+            originAccessControlId: this.originAccessControl.originAccessControlId,
+          },
+        };
+      }
+
+      private getBucketPolicyActions(accessLevels: Set<cloudfront.AccessLevel>) {
+        let actions: string[] = [];
+        for (const accessLevel of accessLevels) {
+          actions = actions.concat(BUCKET_ACTIONS[accessLevel]);
+        }
+        return actions;
+      }
+
+      private getKeyPolicyActions(accessLevels: Set<cloudfront.AccessLevel>) {
+        let actions: string[] = [];
+        for (const accessLevel of accessLevels) {
+          // Filter out DELETE since delete permissions are not applicable to KMS key actions
+          if (accessLevel !== AccessLevel.DELETE) {
+            actions = actions.concat(KEY_ACTIONS[accessLevel]);
+          }
+        }
+        return actions;
+      }
+
+      private grantDistributionAccessToBucket(distributionId: string, actions: string[]): iam.AddToResourcePolicyResult {
+        const oacBucketPolicyStatement = new iam.PolicyStatement(
+          {
+            effect: iam.Effect.ALLOW,
+            principals: [new iam.ServicePrincipal('cloudfront.amazonaws.com')],
+            actions,
+            resources: [bucket.arnForObjects('*')],
+            conditions: {
+              StringEquals: {
+                'AWS:SourceArn': `arn:${Aws.PARTITION}:cloudfront::${Aws.ACCOUNT_ID}:distribution/${distributionId}`,
+              },
+            },
+          },
+        );
+        const result = bucket.addToResourcePolicy(oacBucketPolicyStatement);
+        return result;
+      }
+
+      private grantDistributionAccessToKey(actions: string[], key: IKey): iam.AddToResourcePolicyResult {
+        const oacKeyPolicyStatement = new iam.PolicyStatement(
+          {
+            effect: iam.Effect.ALLOW,
+            principals: [new iam.ServicePrincipal('cloudfront.amazonaws.com')],
+            actions,
+            resources: ['*'],
+            conditions: {
+              ArnLike: {
+                'AWS:SourceArn': `arn:${Aws.PARTITION}:cloudfront::${Aws.ACCOUNT_ID}:distribution/*`,
+              },
+            },
+          },
+        );
+        Annotations.of(key.node.scope!).addWarningV2('@aws-cdk/aws-cloudfront-origins:wildcardKeyPolicyForOac',
+          'To avoid circular dependency between the KMS key, Bucket, and Distribution, ' +
+          'a wildcard is used to match all Distribution IDs in Key policy condition.\n' +
+          'To further scope down the policy for best security practices, see the "Using OAC for a SSE-KMS encrypted S3 origin" section in the module README.');
```

Comment:
```suggestion
          'To avoid a circular dependency between the KMS key, Bucket, and Distribution during the initial deployment, ' +
          'a wildcard is used in the Key policy condition to match all Distribution IDs.\n' +
          'After deploying once, it is recommended to further scope down the policy for best security practices by following the guidance in the "Using OAC for a SSE-KMS encrypted S3 origin" section in the module README.');
```

Changing the wording to make it clearer that it is recommended to further scope down the policy after the initial deployment. 

Reminder to also change the warning in the README along with this.

Reply:
Agreed, will update